sbt/src/sbt-test/actions/compile-clean/test

```diff
@@ -0,0 +1,19 @@
+> test:compile
+$ exists target/scala-2.12/resolution-cache/
+$ exists target/streams/
+$ exists target/scala-2.12/classes/A.class
+$ exists target/scala-2.12/test-classes/B.class
+
+> compile:compile::clean
+$ exists target/scala-2.12/resolution-cache/
+$ exists target/streams/
+$ exists target/scala-2.12/test-classes/B.class
+# it should clean only class files
+$ absent target/scala-2.12/classes/A.class
+
+> test:compile::clean
+$ exists target/scala-2.12/resolution-cache/
+$ exists target/streams/
+# it should clean both compile and test classes
+$ absent target/scala-2.12/classes/A.class
```

Comment:
I have 2 concerns with this test: (1) I think it's over-specifying - it shouldn't make assertions about whether or not there is a `streams` directory, etc, and (2) I also think it's wrong - cleaning test classes shouldn't clean compile classes too.

Reply:
Ideally, only tests' class files should be removed.
  
Can you elaborate how you would implement this so that cleaning test classes shouldn't clean compile classes? I don't think there's a way to avoid that given how sbt deals with configuration dependencies.

Reply:
I don't know, I would have to experiment.

Reply:
> it shouldn't make assertions about whether or not there is a streams directory, etc

@dwijnand OK. What about `resolution_cache`? I just want to check that whatever else is generated by sbt is kept.

Reply:
Same with `resolution_cache`. I agree with the intent. My idea was something along the lines of:

```
test:compile
# sanity check
exists classes
exists test-classes

compile:clean
absent classes
exists test-classes

test:compile

test:clean
exists classes
absent test-classes
```

Reply:
OK. How about creating a file in `target/` and checking that it's not deleted? (because normal `clean` would delete it).

Also I have to check for existence of specific class files, because the root class folders are preserved.

Reply:
> OK. How about creating a file in `target/` and checking that it's not deleted? (because normal `clean` would delete it).

Sounds good.

> Also I have to check for existence of specific class files, because the root class folders are preserved.

Yep - I was just using short-hand.

Reply:
Addressed in 4bba70b0419aff9e50bf0460efed0b5a554ceb4d